### PR TITLE
Debug and Log compilation options

### PR DESCRIPTION
**Pull request by Ircama**
# Debug and Log compilation options

Revised *makefile*, moving the `-g` option to DEBUGMODE and allowing the LOG option.

## Compilation

To perform a standard compilation, set the *RELEASE* flag to *YES*. Check the following example, which also includes prerequisites:

```bash
sudo apt install -y liballegro4.4 liballegro4-dev allegro4-doc
make clean
make -e RELEASE=yes
```

### Enabling logging

In order to enable logging to a file named *"comm_log.txt"* in the current directory, set the *LOG* compilation flag to *YES*, like in the following example:

```bash
make clean
make -e RELEASE=yes LOG=yes
```

### Compilation in DEBUG mode

To compile in debug mode, set the *DEBUGMODE* compilation flag to *YES* instead of using the *RELEASE* flag, like in the following example, which also includes logging to *"comm_log.txt"*:

```bash
make clean
make -e DEBUGMODE=yes LOG=yes
```

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -1,9 +1,11 @@
 CC = gcc
 AL_LIBS = $(shell allegro-config --libs)
-CFLAGS ?= -O2 -g -Wall
+CFLAGS ?= -O2 -Wall
 
 ifdef DEBUGMODE
    DEFINES = -DDEBUG
```
Comment (kees) [inline makefile]: Actually, nothing uses -DDEBUG, so this can be removed entirely.

File: makefile [hunk 2]
```diff
@@ -1,9 +1,11 @@
 CC = gcc
 AL_LIBS = $(shell allegro-config --libs)
-CFLAGS ?= -O2 -g -Wall
```
Comment (kees) [inline makefile:3]: I'd rather this stay the default. Anyone wanting to ship without symbols can strip the binary before release.

File: makefile [hunk 3]
```diff
@@ -36,6 +38,11 @@ ifndef NOWERROR
    CFLAGS += -Werror
 endif
 
+ifdef LOG
+   DEFINES += -DLOG_COMMS
```
Comment (kees) [inline makefile:32]: Actually, I'd much prefer this was a command line option instead of a build option.

**Review by kees — CHANGES_REQUESTED** [review]

File: makefile (hunk 2 above)
Comment (Ircama) [reply makefile:3]: OK, as you want

File: makefile (hunk 3 above)
Comment (Ircama) [reply makefile:32]: Right, might be a future implementation

File: makefile [hunk 4]
```diff
@@ -3,7 +3,6 @@ AL_LIBS = $(shell allegro-config --libs)
 CFLAGS ?= -O2 -Wall
 
 ifdef DEBUGMODE
```
Comment (kees) [inline makefile]: Actually I mean the entire DEBUGMODE build option.

And while we're at it, RELEASE is silly too. I don't either -O3 nor -s. :P

**Review by kees — CHANGES_REQUESTED** [review]

File: makefile (hunk 4 above)
Comment (Ircama) [reply makefile]: Yes, that's right, I noted that too but did not change it assuming that -O2 vs. -O3 had some reason to be used. Notice that -O2 was part of the initial code. What I did instead was simply remove the symbols from the release compilation, confining them to the DEBUGMODE flag. For other improvements, I would recommend a step-by-step approach also assessed on the free time available for the implementation.

**Pull request merged**